examples/ctc.dx

```diff
@@ -30,115 +33,115 @@ def prepend {m v} (first: v) (seq: m=>v) : ((Unit|m)=>v) =
     Left () -> first
     Right i -> seq.i
 
-def prepend_and_interleave {m v} (blank:v) (seq: m=>v) : ((Unit|(m & (Fin 2)))=>v) =
-  -- Turns "text" into " t e x t".
-  -- The output of this function has a slightly complicated output type, which
-  -- has size 1 + 2 * (size m).
-  interleaved = interleave blank seq
-  prepend blank interleaved
-
-def clipidx (n:Type) [Ix n] (i:Int) : n =
-  -- Returns element at 0 if less than zero.
-  -- Ideally we could have an alternative
-  -- to Fin that just clips the index at its bounds.
-  from_ordinal n $ unsafe_i_to_n (select (i < 0) 0 i)
-
-def logaddexp (x:Float) (y:Float) : Float =
-  m = max x y
-  m + ( log ( (exp (x - m) + exp (y - m))))
-
-def ctc {vocab time position} [Eq vocab, Eq position, Eq time]
-      (blank:  vocab)
-      (logits: time=>vocab=>Float)
-      (labels: position=>vocab)
-      : Float =
-  -- Computes log p(labels | logits), marginalizing over possible alignments.
-  -- Todo: remove unnecessary implicit type annotations once
-  -- Dex starts putting implicit types in scope.
+def prepend_and_interleave {m v} (blank:v) (seq: m=>v) : ((Unit|(m & IsBlank))=>v) =
```

Comment:
Not to mess up your architecture here, but do you think this composition might be better served with the `Post` (as in "fence-post") index set from the Prelude?  For an index set `ix`, `Post ix` is the index set of fence-posts around `ix`, supporting functions `left_post : ix -> Post ix` and `right_post : ix -> Post ix`.  Maybe it would make sense to build another index set `FenceAndPosts ix` that iterates over the fences and the posts in the proper order?

Reply:
Oooh, great idea, will try it later today.  I think your will be more readable.